packages/react-app/components/cards/MediaCard.jsx

```diff
@@ -69,140 +83,199 @@ function MediaCard({
+        await tokenContract.approve(dRecruitContract.address, weiStakeAmount);
+      }
+      const tx = await dRecruitContract.request(privateProfile.tokenId, weiStakeAmount, {
+        value: 0,
       });
       toast({
         title: "Request transaction sent",
-        description: <text>Your transaction was successfully sent <a href={`${NETWORKS.mumbai.blockExplorer}/tx/${tx.hash}`} target="_blank" rel="noopener noreferrer"><LinkIcon color="white" /></a></text>,
-        status: "success"
+        description: (
+          <text>
+            Your transaction was successfully sent{" "}
+            <a href={`${NETWORKS.mumbai.blockExplorer}/tx/${tx.hash}`} target="_blank" rel="noopener noreferrer">
+              <LinkIcon color="white" />
+            </a>
+          </text>
+        ),
+        status: "success",
       });
       const receipt = await tx.wait();
       toast({
         title: "Request transaction confirmed",
-        description: <text>Your transaction was confirmed <a href={`${NETWORKS.mumbai.blockExplorer}/tx/${tx.hash}`} target="_blank" rel="noopener noreferrer"><LinkIcon color="white" /></a></text>,
-        status: "success"
-      })
+        description: (
+          <text>
+            Your transaction was confirmed{" "}
+            <a href={`${NETWORKS.mumbai.blockExplorer}/tx/${tx.hash}`} target="_blank" rel="noopener noreferrer">
+              <LinkIcon color="white" />
+            </a>
+          </text>
+        ),
+        status: "success",
+      });
     } catch (err) {
       toast({
         title: "Request transaction failed",
         description: err.message + (err.data ? ` ${err.data.message}` : ""),
-        status: "error"
-      })
+        status: "error",
+      });
     }
+    setStakeAmount();
     // const { data } = await axios.get(`${process.env.NEXT_PUBLIC_API_URL}/unlock/${privateProfile.tokenId}`, {
     //   withCredentials: true,
     // });
     // return data;
   };
 
   return (
-    <Box
-      maxW={"350px"}
-      w={"full"}
-      bg={useColorModeValue("white", "gray.800")}
-      boxShadow={"2xl"}
-      rounded={"3xl"}
-      overflow={"hidden"}
-      my="5"
-    >
-      <Image
-        alt="cover"
-        h={"120px"}
+    <>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Modal Title</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <FormLabel htmlFor="stakeAmount">Enter stake amount in MATIC</FormLabel>
+            <Input
+              value={stakeAmount}
+              onChange={e => setStakeAmount(e.target.value)}
+              placeholder="Minimum 0.1 MATIC"
+              borderColor="purple.500"
+            />
+          </ModalBody>
+
+          <ModalFooter>
+            <Button
+              marginLeft={0}
+              marginRight={"auto"}
+              colorScheme="blue"
+              onClick={() => {
+                onClose();
```

Comment:
Don't close the modal on transaction being sent! We need the context to exist.

Reply:
Bit confused here. The modal doesn't send a transaction. It just captures the stake amount and stores it in a state variable

Reply:
It's a confusing UX pattern for transaction prompts to appear after the modal closes. Functionally, I understand the intent but for an user, it's jarring.

Reply:
I see. I'll work on this next. Won't be able to work on it for dogfooding tonight tho :(